packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,308 @@
+    if (!image.isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+    return sendAndHandleSdkError('permissions.has', permissions);
+  }
+
+  /**
+   * Requests user permission for visualMedia
+   * @beta
+   * @returns Promise that will resolve with true if the user consented permission for media(including Camera and Gallery permission), or with false otherwise,
+   * In case of an error, promise will reject with the error.
+   * @throws NOT_SUPPORTED_ON_PLATFORM Error if the DevicePermission.Media permission has not successfully granted.
+   */
+  export function requestPermission(): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!image.isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+    return sendAndHandleSdkError('permissions.request', permissions);
+  }
+
+  /**
+   * To enable this image capability will let the app developer ask the user to get images from camera/local storage
+   * @beta
+   */
+  export namespace image {
+    /**
+     * CameraImageProperties is for the image taken from the camera
+     * @beta
+     */
+    export interface CameraImageProperties extends VisualMediaProps {
+      /**
+       * The source in CameraImageProperties should always be CameraProps
+       */
+      sourceProps: CameraProps;
+
+      /**
+       * The common additional properties for image in mobile devices
+       */
+      commonImageprops: ImageProperties;
+    }
+
+    /**
+     * CameraImageProperties is for the image taken from the camera
+     * @beta
+     */
+    export interface GalleryImageProperties extends VisualMediaProps {
+      /**
+       * The source in GalleryImageProperties should always be GalleryProps
+       */
+      sourceProps: GalleryProps;
+
+      /**
+       * The common additional properties for image in mobile devices
+       */
+      commonImageprops: ImageProperties;
+    }
+
+    /**
+     *  All properties in ImageProperties are optional and have default values in the platform
+     *  Additional properties for image in mobile devices
+     * @beta
+     */
+    export interface ImageProperties {
+      /**
+       * Optional; indicate if inking on the selected Image is allowed or not
+       * Default value is false
+       */
+      shouldAllowInkingOnImages?: boolean;
+
+      /**
+       * Optional; indicate if putting text stickers on the selected Image is allowed or not
+       * Default value is false
+       */
+      shouldAllowTextStickersOnImages?: boolean;
+
+      /**
+       * Optional; indicate if the filter mode is enabled on the selected image
+       * Default value is false
+       */
+      enableFilterOnImages?: boolean;
+
+      /**
+       * Optional; Lets the developer specify the image output formats, more than one can be specified.
```

Comment:
>Lets the developer specify the image output formats, [](http://example.com/codeflow?start=19&length=52)

Could this be clarified? E.g., "Specifies which formats the user is allowed to choose"

Reply:
Thanks for your suggestion, updated!